src/App.tsx

```diff
@@ -1,14 +1,82 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-nested-ternary */
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilteredBy } from './types/Filter';
+import { getTodos } from './api';
+
+// const filteredTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
+//   let readyTodos = [...todos];
+
+//   if (query) {
+//     const readyQuery = query.toLowerCase().trim();
+
+//     readyTodos = readyTodos.filter(todo => todo.title.toLowerCase()
+//       .includes(readyQuery));
+//   }
+
+//   switch (filterBy) {
+//     case FilteredBy.ACTIVE:
+//       return readyTodos.filter(todo => !todo.completed);
+//     case FilteredBy.COMPLETED:
+//       return readyTodos.filter(todo => todo.completed);
+//     default:
+//       return readyTodos;
+//   }
+// };
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
+  const readyQuery = query.toLowerCase().trim();
+
+  const visibleTodos = (filterBy === FilteredBy.ACTIVE)
+    ? todos.filter(todo => !todo.completed)
+    : (filterBy === FilteredBy.COMPLETED
+      ? todos.filter(todo => todo.completed)
+      : [...todos]);
+
+  return readyQuery
+    ? visibleTodos.filter(todo => todo.title.toLowerCase().includes(readyQuery))
+    : visibleTodos;
+};
```

Comment:
it's better not use nested ternary operator. 
You can use `switch` statement

Example: 
```
const filteredTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
  switch (filterBy) {
    case FilteredBy.ACTIVE:
      return todos.filter(todo => !todo.completed)
    case FilteredBy.COMPLETED:
      return todos.filter(todo => todo.completed)
    default:
      return  [...todos]
  }
};
```